docs/reference/remap/functions/redact.cue

```diff
@@ -0,0 +1,94 @@
+package metadata
+
+remap: functions: redact: {
+	category: "String"
+	description: """
+		Redact potentially sensitive data in `value` such as:
+
+		- credit card numbers
+		- custom patterns
+		- (more to come!)
+
+		This can act as a last resort in the case your applications log sensitive data by accident to ensure you do not
+		leak this information into your log storage.
+		"""
+
+	arguments: [
+		{
+			name: "value"
+			description: #"""
+				The value to redact sensitive data from.
+
+				Its behavior differs depending on the type of `value`:
+				- For strings, it simply redacts the sensitive data and returns a new string
+				- For arrays, it redacts the sensitive data in each string element
+				- For objects, it masks the sensitive data in each string value, but not keys
+
+				For arrays and objects it will recurse into any nested arrays or objects. Any non-string elements will
+				be skipped.
+
+				Any redacted text will be replaced with `[REDACTED]`.
+				"""#
+			required: true
+			type: ["string", "object", "array"]
+		},
+		{
+			name: "filters"
+			description: #"""
+					List of filters to be applied to the `value`.
+
+					If none, the default set of filters is applied: ["credit_card"].
```

Comment:
We should callout why someone would choose the default to align on intentions. For example, a customer could choose the default to delegate PII redaction maintenance to Vector. This will be limited to redimentary/high-confidence redactions. More sophisticated redaction will require rules, etc, something we are teasing out in the enterprise version.

cc @zsherman

Reply:
Yeah, this comment also made me think maybe this PR is biting off more than it should. I think long term, this is a good idea to have a default suite of filters and let users just delegate to vector to Do the Right Thing, but I don't think we are ready yet to make that decision for them so I pushed 56c8508 removing the default filters so users will explicitly opt into the ones they want.